doc/github-development-workflow.md

```diff
@@ -36,10 +36,10 @@ requests.
 MUST be submitted as a pull request to GitHub.  All changes to the
 "develop" branch must be made exclusively through merging pull requests.
 The "release" and "stable" branches, respectively, are only to be
-updated upon feature or stable releases based on the associated
-tags.  Updates to the stable release in between stable releases
+updated upon "feature releases" or "stable releases" based on the
+associated tags.  Updates to the stable release in between stable releases
```

Comment:
`associated` tags is not clear IMO. I assume you mean `patch_` and `stable_` tags? Maybe you should document all this in 1 place?

Reply:
This is not a file meant to be read by regular uses, but only by LAMMPS developers. Those know what the tags mean.

Documenting "everything in one place" is not a good idea since there are different "consumers" of the documentation: regular users, users that make modifications to the source, LAMMPS developers. In particular for the first kind, there needs to be some redundancy, since those rarely follow links and thus some minimal information has to be put right where they need it.